include/ginkgo/core/factorization/incompleted_factorization.hpp

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_FACTORIZATION_INCOMPLETED_FACTORIZATION_HPP_
+#define GKO_PUBLIC_CORE_FACTORIZATION_INCOMPLETED_FACTORIZATION_HPP_
+
+
+namespace gko {
+namespace factorization {
+
+
+/**
+ * An enum class for algorithm selection in the incomplete factorization.
+ * `sparselib` is only available for CUDA and HIP.
+ * `syncfree` is Ginkgo's implementation through the Lu/Cholesky factorization
+ * with given sparsity.
+ */
+enum class factorize_algorithm { sparselib, syncfree };
+
+
```

Comment:
Isn't it confusing for this to be in its own incomplete factorization file but still just be in the `factorization` namespace and be named `factorize_algorithm`? Nothing in the name `gko::factorization::factorize_algorithm` indicates that it should be in its own file dedicated to incomplete factorizations.

Reply:
I do not have a good idea other than `incomplete_factorize_algorithm`.
It is a bit long but should be descriptive.

Reply:
How about `factorization::incomplete_algorithm`? Otherweise factorize would be redundant

Reply:
I like that. Or what about another namespace: `factorization::incomplete::algorithm`?

Reply:
I currently went for `incomplete_algorithm`.
I do not have strong preference on both.